static/js/components/UploadPhotometry.jsx

```diff
@@ -0,0 +1,385 @@
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useParams, Link } from "react-router-dom";
+import MUIDataTable from "mui-datatables";
+import TextareaAutosize from "@material-ui/core/TextareaAutosize";
+import Select from "@material-ui/core/Select";
+import MenuItem from "@material-ui/core/MenuItem";
+import Button from "@material-ui/core/Button";
+import InputLabel from "@material-ui/core/InputLabel";
+import FormControl from "@material-ui/core/FormControl";
+import Card from "@material-ui/core/Card";
+import CardContent from "@material-ui/core/CardContent";
+import Box from "@material-ui/core/Box";
+import Tooltip from "@material-ui/core/Tooltip";
+import HelpOutlineIcon from '@material-ui/icons/HelpOutline';
+import Typography from "@material-ui/core/Typography";
+import { makeStyles, withStyles } from "@material-ui/core/styles";
+import { useForm, Controller } from "react-hook-form";
+
+import FormValidationError from "./FormValidationError";
+import * as Actions from "../ducks/source";
+
+
+const textAreaPlaceholderText = `mjd,flux,fluxerr,zp,magsys,instrument_id,filter,altdata.meta1
+58001.,22.,1.,30.,ab,1,ztfg,44.4
+58002.,23.,1.,30.,ab,1,ztfg,43.1
+58003.,22.,1.,30.,ab,1,ztfg,42.5`;
+
+const HtmlTooltip = withStyles((theme) => ({
+  tooltip: {
+    backgroundColor: '#f9f9ff',
+    color: 'rgba(0, 0, 0, 0.87)',
+    maxWidth: 700,
+    fontSize: theme.typography.pxToRem(12),
+    border: '1px solid #dadde9',
+  },
+}))(Tooltip);
+
+const UploadPhotometryForm = () => {
+  const dispatch = useDispatch();
+  const { instrumentList } = useSelector((state) => state.instruments);
+  const [showPreview, setShowPreview] = useState(false);
+  const [csvData, setCsvData] = useState({});
+  const [successMessage, setSuccessMessage] = useState("");
+  const { id } = useParams();
+  const { handleSubmit, errors, reset, control, getValues } = useForm();
+  let formState = getValues();
+
+  const validateCsvData = () => {
+    setShowPreview(false);
+    setCsvData({});
+    setSuccessMessage(null);
+    formState = getValues();
+    if (!formState.csvData) {
+      return "Missing CSV data";
+    }
+    const delim = new RegExp(formState.delimiter);
+    let [header, ...dataRows] = formState.csvData.trim().split("\n");
+    header = header.split(delim);
+    dataRows = dataRows.map((row) => row.split(delim));
```

Comment:
[CSV files](https://tools.ietf.org/html/rfc4180) may contain commas, so we need a slightly more sophisticated splitting strategy.

Here's a [code snippet](https://gist.github.com/plbowers/7560ae793613ee839151624182133159) that we can use more-or-less as-is.

Reply:
(If we want to do it beautifully, we can even generate a parser using a PEG specification: http://jsfiddle.net/knvzk/10).  We'd only generate the parser once and then re-use it, not generate it on the fly for each user.